### PR TITLE
Autocorrelation check for event_optimize

### PR DESCRIPTION
**Pull request by devbhakt**
I have an autocorrelation routine for `event_optimize`. The routine runs the sampler and determines convergence of the chains by 2 criteria:

1.  The chain length is a specified ratio times the estimated autocorrelation time for each parameter.  (The default is set to 10 in `event_optimize`)
2. The change in estimated autocorrelation times is less than 1%

The check is conducted every 100 steps and then every 25 steps once 10% convergence is reached.

### CODE REVIEW

File: src/pint/scripts/event_optimize.py [hunk 1]
```diff
@@ -545,6 +621,12 @@ def main(argv=None):
         default=False,
         action="store_true",
     )
+    parser.add_argument(
+        "--autocorr",
+        help="Turn the autocorrelation check function off",
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:626]: What is the behavior of event_optimize when "--autocorr" is given? If it is to turn off the autocorrelation, I think this option will be better called "--no-autocorr".

Comment (devbhakt) [reply src/pint/scripts/event_optimize.py:626]: I've updated the option now

File: src/pint/scripts/event_optimize.py [hunk 2]
```diff
@@ -247,6 +247,82 @@ def get_fit_keyvals(model, phs=0.0, phserr=0.1):
...
+    The criteria to check for convergence is:
+        1. the chain has to be longer than the specified ratio times the estimated autocorrelation time
+        2. the change in the estimated autocorrelation time is less than 1%
+    """
+    autocorr = []
+    old_tau = np.inf
+    converged1 = False
+    converged2 = False
+    for sample in sampler.sample(pos, iterations=nsteps, progress=True):
+        if converged1 == False:
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:279]: if not converged1

File: src/pint/scripts/event_optimize.py [hunk 3]
```diff
@@ -247,6 +247,82 @@ def get_fit_keyvals(model, phs=0.0, phserr=0.1):
...
+                        log.info(
+                            "10 % convergence reached with a mean estimated integrated step: "
+                            + str(x)
+                        )
+                    else:
+                        continue
+            else:
+                continue
+        else:
+            if converged2 == False:
```
Comment (abhisrkckl) [inline src/pint/scripts/event_optimize.py:302]: if not converged2

**Pull request merged**